src/models/companyHouseEntity.ts

```diff
@@ -0,0 +1,75 @@
+import { injectable, singleton } from 'tsyringe'
+import { z } from 'zod'
+import { Env } from '../env'
+
+const companyProfileSchema = z.object({
+  company_number: z
+    .string()
+    .regex(
+      new RegExp(
+        /^((AC|ZC|FC|GE|LP|OC|SE|SA|SZ|SF|GS|SL|SO|SC|ES|NA|NZ|NF|GN|NL|NC|R0|NI|EN|\d{2}|SG|FE)\d{5}(\d|C|R))|((RS|SO)\d{3}(\d{3}|\d{2}[WSRCZF]|\d(FI|RS|SA|IP|US|EN|AS)|CUS))|((NI|SL)\d{5}[\dA])|(OC(([\dP]{5}[CWERTB])|([\dP]{4}(OC|CU))))$/
+      )
+    ),
+  company_name: z.string(),
+  registered_office_address: z.object({
+    address_line_1: z.string().optional(),
+    address_line_2: z.string().optional(),
+    care_of: z.string().optional(),
+    country: z.string().optional(),
+    locality: z.string().optional(),
+    po_box: z.string().optional(),
+    postal_code: z.string().optional(),
+    premises: z.string().optional(),
+    region: z.string().optional(),
+  }),
+  registered_office_is_in_dispute: z.boolean(),
+  company_status: z.union([
+    z.literal('active'),
+    z.literal('dissolved'),
+    z.literal('liquidation'),
+    z.literal('receivership'),
+    z.literal('converted-closed'),
+    z.literal('voluntary-arrangement'),
+    z.literal('insolvency-proceedings'),
+    z.literal('administration'),
+    z.literal('open'),
+    z.literal('closed'),
+    z.literal('registered'),
+    z.literal('removed'),
+  ]),
+})
+export type CompanyProfile = z.infer<typeof companyProfileSchema>
+
+@singleton()
+@injectable()
+export default class CompanyHouseEntity {
+  constructor(private env: Env) {}
+
+  private async makeCompanyProfileRequest(route: string): Promise<unknown> {
+    const url = new URL(route)
+
+    const response = await fetch(url.toString(), {
+      method: 'GET',
+      headers: new Headers({
+        Authorization: 'Basic ' + process.env.COMPANY_PROFILE_API_KEY,
+      }),
+    })
+
+    if (!response.ok) {
+      throw new Error(`Error calling CompanyHouse API`)
```

Comment:
once we have more Errors we can abstract them like we do elsewhere -> https://github.com/digicatapult/sqnc-hyproof-api/blob/main/src/lib/error-handler/index.ts

Reply:
We already have https://github.com/digicatapult/veritable-ui/blob/main/src/errors.ts. This would be an `InternalError`